tls/s2n_handshake_io.c

```diff
@@ -441,6 +443,47 @@ int s2n_conn_set_handshake_no_client_cert(struct s2n_connection *conn) {
     return 0;
 }
 
+const char *s2n_connection_get_handshake_type_name(struct s2n_connection *conn) 
+{
+    int handshake_type = conn->handshake.handshake_type;
+    
+    if (handshake_type_str[handshake_type][0] != '\0') {
+        return handshake_type_str[handshake_type];
+    }
+
+    /* Compute handshake_type_str[handshake_type] */
+    if (handshake_type > INITIAL) {
+        if (handshake_type & NEGOTIATED) {
+            strcat(handshake_type_str[handshake_type], "NEGOTIATED|");
```

Comment:
`strncat` to avoid buffer overrun in case we forget to update the limit?

Reply:
> `strncat` to avoid buffer overrun in case we forget to update the limit?

Done